pkg/mount/block/blockdev.go

```diff
@@ -498,6 +498,32 @@ func (b BlockDevices) FilterPartType(guid string) BlockDevices {
 	return b.FilterNames(names...)
 }
 
+// FilterPartLabel returns a list of BlockDev objects whose underlying block
+// device has the given parition label.
+func (b BlockDevices) FilterPartLabel(label string) BlockDevices {
+	var names []string
+	for _, device := range b {
+		table, err := device.GPTTable()
+		if err != nil {
+			continue
+		}
+		for i, part := range table.Partitions {
+			if part.IsEmpty() {
+				continue
+			}
+			if strings.ToLower(part.Name()) == strings.ToLower(label) {
+				r := []rune(device.Name[len(device.Name)-1:])
```

Comment:
Won't this break once you go beyond 10 partitions?

Reply:
Oh, that's true. Should this case be covered. Or should we add a comment. Same goes for FilterPartID and FilterPartType.

Reply:
At minimum skip the partition if the number is >10. Right now, it looks like it'll try to boot from the wrong partition in this case.

Reply:
I've moved that part into a helper function and added a test. Seems to work out with higher partition numbers. If I miss anything here, let me know.